File: src/checkers-api/Models/Responses/GameInfo.cs
using checkers_api.Models.GameModels;

namespace checkers_api.Models.Responses;

public class GameInfo
{
    public string RoomId { get; }
    public bool IsGameOver { get; }
    public Player NextPlayerTurn { get; }
    public Player? Winner { get; }
    public List<List<Piece?>> Board { get; }

    public GameInfo(string roomId, Player nextPlayerTurn, List<List<Piece?>> board, Player? winner = null)
    {
        RoomId = roomId;
        NextPlayerTurn = nextPlayerTurn;
        Board = board;
        Winner = winner;
        IsGameOver = winner is not null;
    }
}

File: src/checkers-client/src/models/game/gameInfo.ts
import Piece from "./piece";
import Player from "./player";

export default class GameInfo {
  public readonly roomId: string;
  public readonly isGameOver: boolean;
  public readonly nextPlayerTurn: Player;
  public readonly board: Piece[][];
  public readonly winner?: Player;

  constructor(
    roomId: string,
    isGameOver: boolean,
    nextPlayerTurn: Player,
    board: Piece[][],
    winner?: Player
  ) {
    this.roomId = roomId;
    this.isGameOver = isGameOver;
    this.nextPlayerTurn = nextPlayerTurn;
    this.board = board;
    this.winner = winner;
  }
}
